src/blog/2024/01/send-a-file.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Send a File to Node-RED
+subtitle: A guide to sending a csv file to Node-RED and start interacting with it.
+date: 2024-01-03
+authors: ["grey-dziuba"]
+image: /blog/2023/12/images/uns-data-modeling.png
+tags:
+    - posts
+    - flowfuse
+    - how-to
+---
+
+Have you ever needed to send a CSV file to your Node-RED instance? This file can go on to populate a shift schedule, product specifications, or some other configuration file that is used. In this guide, we are leveraging a Python script that runs a Flask application to post the file to a Node-RED instance and organize the data to be sent on or used.
+<!--more-->
+
+## Why would you need to send a file to Node-RED?
+
+Often times it is necessary to update lookup tables in a SQL database, but you don't necessarily want to give access to everyone to edit the database, nor do you want to have to do it all yourself. This can often be seen when new products are introduced into a manufacturing facility. It may not be often, but enough that it warrants its own application. This process will guide you in a way that will enable your teammates to upload the files to the system themselves.
+
+## Summary
+
+There are two approaches that were taken when solving this. The first approach is a simple script that will be shared below. It is a simple Python application that allows the user to send a file with a simple command, but this might require a little more technical skills that the end user may not feel comfortable with. Which brought us to pathway number two. A web-based application that allows the user to upload files to a browser with a selectable endpoint. Both have their pros and cons. I will let you decide which is best for your application.
```

Comment:
Method one: `curl -X POST -d @/path/to/file.csv` seems very simple. For solution two; can we create a dashboard that allows for file uploads? (cc @joepavitt )

Reply:
We have an issue for it as a core node: https://github.com/FlowFuse/node-red-dashboard/issues/249 although did also have someone that used `ui-template` to do it too: https://github.com/FlowFuse/node-red-dashboard/issues/369

Reply:
I am including both curl and a Dashboard input along the other two options.